tools/tpm2_nvdefine.c

```diff
@@ -201,27 +200,47 @@ static tool_rc handle_no_index_specified(ESYS_CONTEXT *ectx, TPM2_NV_INDEX *chos
     UINT32 count = capabilities->data.tpmProperties.count;
 
     if (!count) {
-        LOG_ERR("Could not get maximum NV index, try specifying an NV index");
+        LOG_ERR("Could not get maximum NV index size");
         goto out;
     }
 
     UINT32 i;
     for (i=0; i < count; i++) {
         if (properties[i].property == TPM2_PT_NV_INDEX_MAX) {
-            max = TPM2_HR_NV_INDEX | properties[i].value;
+		*size = properties[i].value;
+		break;
         }
     }
 
-    if (!max) {
+    if (*size == 0) {
         LOG_ERR("Could not find max NV indices in capabilities");
         goto out;
     }
 
-    /* done getting max NV index */
+    rc = tool_rc_success;
+
+out:
     free(capabilities);
-    capabilities = NULL;
+
+    return rc;
+}
+
+static tool_rc handle_no_index_specified(ESYS_CONTEXT *ectx, TPM2_NV_INDEX *chosen) {
+
+    tool_rc rc = tool_rc_general_error;
+
+    TPM2_NV_INDEX max = 0;
+    UINT16 nv_max_sz = 0;
+
+    tool_rc tmp_rc = get_max_nv_index_size(ectx, &nv_max_sz);
+    if (tmp_rc != tool_rc_success) {
+        return tmp_rc;
+    }
```

Comment:
You shouldn't need to modify handle_no_index_specified(). It's just looking up the first free NV index, it has nothing to do with size.

Reply:
Ok, will skip this patch.